packages/ui/src/forum/types/ForumCategory.ts

```diff
@@ -70,13 +70,17 @@ export const asForumCategory = (fields: ForumCategoryFields): ForumCategory => (
   ...asBaseForumCategory(fields),
   moderators: fields.moderators?.map(({ id, membership }) => ({ id, handle: membership.handle })) ?? [],
   subcategories:
-    fields.forumcategoryparent?.map((fields) => ({ ...asSubCategory(fields), status: fields.status.__typename })) ?? [],
+    fields.forumcategoryparent
+      ?.filter((fields) => fields.status.__typename !== 'CategoryStatusRemoved')
+      .map((fields) => ({ ...asSubCategory(fields), status: fields.status.__typename })) ?? [],
 })
 
 export const asArchivedForumCategory = (fields: ArchivedForumCategoryFields): ForumCategory => ({
   ...asBaseArchivedForumCategory(fields),
   subcategories:
-    fields.forumcategoryparent?.map((fields) => ({ ...asSubCategory(fields), status: fields.status.__typename })) ?? [],
+    fields.forumcategoryparent
+      ?.filter((subcategory) => subcategory.status.__typename !== 'CategoryStatusRemoved')
```

Comment:
`CategoryStatusRemoved` is supposed to propagate down to subcategories. So when a categories gets archive, it's subcategories should be archived too.

E.g on `query.joystream.org`:
![image](https://user-images.githubusercontent.com/6571453/185429869-4d3a6e9d-d7ef-4ae6-bea7-ca9c3c9ba87b.png)

So here we don't want to filter the `CategoryStatusRemoved`